libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -2673,12 +2696,14 @@ CK_DECLARE_FUNCTION( CK_RV, C_GetAttributeValue )( CK_SESSION_HANDLE hSession,
         {
             xClass = CKO_PUBLIC_KEY;
         }
-        else
+        else if( 0 == mbedtls_x509_crt_parse( &xMbedX509Context, pxObjectValue, ulLength ) )
         {
-            /* TODO: Do we want to safety parse the cert?
-             * Assume certificate. */
             xClass = CKO_CERTIFICATE;
         }
+        else
+        {
+            /* Could not determine the object class." */
```

Comment:
1. Is this an error case?  If so, should you set an error in `xReturn`, e.g. `            xResult = CKR_OBJECT_HANDLE_INVALID;`?
1. There is an extraneous `"` at the end of the comment on line 2705.

Reply:
Done